### PR TITLE
global: repository syncing issues fix

### PR DESCRIPTION
**Pull request by krzysztof**
Signed-off-by: Krzysztof Nowak <k.nowak@cern.ch>

### CODE REVIEW

**Comment by jirikuncar** [comment]
@krzysztof please remove the release commit.

**Comment by lnielsen** [comment]
Any particular reason to remove it? We are unfortunately in need of some rapid release cycles....

**Comment by lnielsen** [comment]
Which additional pre-release checks?

File: invenio_github/api.py [hunk 1]
```diff
@@ -176,9 +175,12 @@ def _sync_hooks(self, repos, async=True):
...
             for repo_id in repos:
-                with db.session.begin_nested():
-                    self.sync_repo_hook(repo_id)
-                db.session.commit()
+                try:
+                    with db.session.begin_nested():
+                        self.sync_repo_hook(repo_id)
+                    db.session.commit()
+                except (NoResultFound, RepositoryAccessError):
+                    pass
```
Comment (lnielsen) [inline invenio_github/api.py]: ``current_app.logger.warning("....", exc_info=True)``

File: invenio_github/tasks.py [hunk 2]
```diff
@@ -75,11 +76,14 @@ def sync_hooks(user_id, repositories):
...
-            # the user might enable/disable a hook.
-            db.session.commit()
+            try:
+                with db.session.begin_nested():
+                    gh.sync_repo_hook(repo_id)
+                # We commit per repository, because while the task is running
+                # the user might enable/disable a hook.
+                db.session.commit()
+            except (NoResultFound, RepositoryAccessError):
+                pass
```
Comment (lnielsen) [inline invenio_github/tasks.py]: ``current_app.logger.warning("....", exc_info=True)``

**Comment by tiborsimko** [comment]
The release commit should be in a separate PR for many reasons in my eyes: 
- The bug fix or a feature request may fail and/or may need review amendments,  which is easier to do without release commits on top. *(tactical reason*)
- The maintainer can take advantage of the release to add other minor things, or say update I18N translations if applicable. Basically, it's up to the maintainer to decide about best time to release. *(strategical reason)*
- "Doing one thing at the time and doing it well."  Are we concentrating on making a nice bug fix or a nice feature implementation in the PR?  Or are we focusing on the release notes, the data migration procedures, the integration checks, etc? *(focus reason)*
- A PR in separate branch would make it easier to rebase on top of any other changes that may come before release. Or imagine if the release fails because Travis tag pushing fails  for the maintainer (happened a few times). It's not about making release days later, it could be simply 30 minutes later, after I18N phrases come. *(practical reason)*
- To make a release after every minor change asked by every service using the package would be definitely not a scalable model for the bright stable future. The lambda users would hate us because they cannot follow news stream, and PyPI.org sysadmins would hate us because we would be eating all their quotas. :)
   *(looking out for a stable future model; I know this case is different...)*
- Let's try to come with a scalable model, if a package is used by many services, because it wouldn't scale to make a new release after every commit. For example, one could imagine services relying on a pinned SHA1 from `master` branch, if they need to deploy quickly after a merge was done. These SHA1 never change, and the model would be more scalable with Ns services and Np patches?! *(looking out for a stable future model again)* 

*Disclaimer:* I was trying to muse about a scalable release model vs a scalable deployment model for the stable times in a generic use case when a package is used by Nu services,  contributed to by Nc people, and maintained by Nm people. I know that in this case things are different, the package is still alpha, and still not used beyond Zenodo... but perhaps it's time to come with scalable practices to pave the way for brighter tomorrows? Let's muse IRL...

**Comment by lnielsen** [comment]
@tiborsimko I will merge once green, and we will come talk to you IRL.

**Comment by david-caro** [comment]
@tsimko What abut automatic semantic versioning and releases on tags?
That would allow getting rid of all and any release prs (as any merged commit would be good and ready to be released) and will still allow the maintainer to decide what and when to release, in a very easy way, just by pushing a tag to the repo.
 And also, as the semantic versioning scheme is kept, it allows any service manager to pin the version to the major number to avoid non-backwards compatible upgrades, having time to properly test upgrades in staging before deploying, and still getting bugfixes and compatible features.
This leaves open the ability to fast-track a change (as the release would be just pushing a tag) if needed, and a more continuous and stable flow of changes otherwise. And get rid of a bunch of manual work while addressing all the issues you raised.

**Pull request merged**